climakitae/selectors.py

```diff
@@ -334,38 +424,114 @@ class DataSelector(param.Parameterized):
         default="monthly", objects=["hourly", "daily", "monthly"]
     )  # for WRF, will just coarsen data to start
 
-    # not needed yet until we have LOCA data:
-    # dyn_stat = param.ListSelector(objects=['Dynamical','Statistical'])
-
-    # @param.depends('timescale','dyn_stat', watch=True)
-    # def _update_variables(self):
-    #    '''
-    #    Updates variable choices, which are different between dynamical/statistical, and
-    #    for statistical are also different for hourly/daily.
-    #    '''
-    #    variables = choices._variable_choices[self.timescale][self.dyn_stat]
-    #    self.param['variable'].objects = variables
-    #    self.variable = variables[0]
+    time_slice = param.Range(default=(1980, 2015), bounds=(1950, 2100))
 
     scenario = param.ListSelector(
-        objects=list(_choices._scenarios["45 km"].keys()), allow_None=True
+        default=["Historical Climate"],
+        objects=list(_choices._scenarios["45 km"].keys()),
+        allow_None=True,
     )
     resolution = param.ObjectSelector(default="45 km", objects=_choices._resolutions)
     append_historical = param.Boolean(default=False)
 
-    @param.depends("resolution", "append_historical", watch=True)
+    @param.depends("resolution", "append_historical", "scenario", watch=True)
     def _update_scenarios(self):
+        """
+        The scenarios available will depend on the resolution (more will be available for 9km
+        than 3km for WRF eventually). Also ensures that "Historical Climate" is not
+        redundantly displayed when "Append historical" is also selected.
+        """
         _list_of_scenarios = list(self._choices._scenarios[self.resolution].keys())
-        if self.append_historical:
-            if "Historical Climate" in _list_of_scenarios:
-                _list_of_scenarios.remove("Historical Climate")
         self.param["scenario"].objects = _list_of_scenarios
-        self.scenario = [
-            None
-        ]  # resetting this avoids indexing errors with prior values
+        if self.append_historical and self.scenario is not None:
+            if "Historical Climate" in self.scenario:
+                _scenarios = self.scenario
+                _scenarios.remove("Historical Climate")
+                self.scenario = _scenarios
+
+    @param.depends("scenario", "append_historical", watch=True)
+    def _update_time_slice_range(self):
```

Comment:
As a general comment, it would be good to think about how to read the available year ranges from the catalog instead of hard coding them here, which will be problematic if LOCA uses a different historical period.

Reply:
Agreed, except that LOCA is being trained on this same WRF data, so the years will be the same. Will make a card about going back and generalizing it though.